application/app/Policies/InstitutionUserPolicy.php

```diff
@@ -0,0 +1,109 @@
+    {
+        return $this->isCurrentUser($institutionUser)
+            || Auth::hasPrivilege(PrivilegeKey::EditUser->value)
+            && $this->isInSameInstitutionAsCurrentUser($institutionUser);
+    }
+
+    public function delete(JwtPayloadUser $jwtPayloadUser, InstitutionUser $institutionUser): bool
+    {
+        throw new BadMethodCallException();
+    }
+
+    public function restore(JwtPayloadUser $jwtPayloadUser, InstitutionUser $institutionUser): bool
+    {
+        throw new BadMethodCallException();
+    }
+
+    public function forceDelete(): bool
+    {
+        throw new BadMethodCallException();
+    }
+
+    public function export(): bool
+    {
+        return Auth::hasPrivilege(PrivilegeKey::ExportUser->value);
+    }
+
+    public function isCurrentUser(InstitutionUser $institutionUser): bool
+    {
+        return filled($currentUserId = Auth::user()?->id)
+            && $currentUserId === $institutionUser->id;
+    }
+
+    public function isInSameInstitutionAsCurrentUser(InstitutionUser $institutionUser): bool
+    {
+        return filled($currentInstitutionId = Auth::user()?->institutionId)
+            && $currentInstitutionId === $institutionUser->institution_id;
+    }
+
+    // Should serve as a query enhancement to Eloquent queries
+    // to filter out objects that the user does not have permissions to.
+    //
+    // Example usage in query:
+    // Role::getModel()->withGlobalScope('policy', RolePolicy::scope())->get();
+    //
+    // The 'policy' string in the example is not strict and is used internally to identify
+    // the scope applied in Eloquent querybuilder. It can be something else as well,
+    // but it should correspond with the intentions of the scope. Using 'policy' provides
+    // general understanding throughout the whole project that the applied scope is related to policy.
+    // The withGlobalScope method does not apply the scope globally, it applies to only the querybuilder
+    // of current query. The method name could be different, but in the sake of reusability
+    // we can use this method that's provided by Laravel and used internally.
+    //
+    public static function scope(): InstitutionUserScope
+    {
+        return new Scope\InstitutionUserScope();
+    }
+}
+
+// Scope resides in the same file with Policy to enforce scope creation with policy creation.
+
+namespace App\Policies\Scope;
+
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Scope as IScope;
+use Illuminate\Support\Facades\Auth;
+
+class InstitutionUserScope implements IScope
+{
+    public function apply(Builder $builder, Model $model): void
+    {
+        if (empty($currentUserInstitutionId = Auth::user()?->institutionId)) {
+            abort(401);
+        }
+
+        $builder->where('institution_id', $currentUserInstitutionId);
+    }
+}
```

Comment:
@DanBondarenko I would like propose to moving this class to a separate file that can be placed inside the `Policies/Scopes` directory. It's a common approach to place each class into a separate file as defined in the PSR-4 autoloading standard which maps namespaces to the directory structure of the application.

Reply:
@kadmit Yeah, that makes sense – I am not opposed. However, since the stub was written by, I guess we should confirm with him. @thenouan, is it OK if we move the policy scope into a separate file (and do so in future as well)?

Reply:
I'm agreeing with it, but I would not do it. The reason for having it in the same class is described in the comment that's above it.

> // Scope resides in the same file with Policy to enforce scope creation with policy creation.

The easiest way to enforce it was to extend the policy creation stub. Other way would have been to create a new command for policy scope boilerplate creation, but that would have been more work